src/test/java/com/infobip/jira/JiraVersionGeneratorHookTest.java

```diff
@@ -1,317 +1,339 @@
+        given(olderRefChange.getRef()).willReturn(olderMinimalRef);
+        given(olderMinimalRef.getId()).willReturn(branchName);
+    }
 
-		given(latestRefChange.getToHash()).willReturn(hash);
-		given(latestRefChange.getRefId()).willReturn(branchName);
-	}
+    private void givenLatestRefChange(String hash, String branchName) {
 
-	private void givenChangesets(Commit... changesets) {
+        given(latestRefChange.getToHash()).willReturn(hash);
+        given(latestRefChange.getRef()).willReturn(latestMinimalRef);
+        given(latestMinimalRef.getId()).willReturn(branchName);
+    }
 
-		given(changesetPage.getValues()).willReturn(ImmutableList.copyOf(changesets));
-		given(historyService.getCommits(any(CommitsRequest.class),
-		                                   any(PageRequest.class))).willReturn(changesetPage);
-	}
+    private void givenChangesets(Commit... changesets) {
 
-	private void givenRepositoryName(String value) {
+        given(changesetPage.getValues()).willReturn(ImmutableList.copyOf(changesets));
+        given(historyService.getCommits(any(CommitsRequest.class),
+                any(PageRequest.class))).willReturn(changesetPage);
+    }
 
-		given(repository.getName()).willReturn(value);
-	}
+    private void givenRepositoryName(String value) {
 
-	private void whenPostReceive(RefChange refChange) {
+        given(repository.getName()).willReturn(value);
+    }
 
-		jiraVersionGeneratorHook.postReceive(repositoryHookContext, ImmutableList.of(refChange));
-	}
+    private void whenPostReceive(RefChange refChange) {
 
-	private void thenShouldAddVersionToIssues(Version version,
-	                                          IssueKey... issueKeys) throws CredentialsRequiredException, ResponseException, JsonProcessingException {
+        jiraVersionGeneratorHook.postReceive(repositoryHookContext, ImmutableList.of(refChange));
+    }
 
-		then(jiraService).should().addVersionToIssues(version, ImmutableList.copyOf(issueKeys));
-	}
+    private void thenShouldAddVersionToIssues(String versionName,
+                                              ProjectKey projectKey,
+                                              IssueKey... issueKeys) throws CredentialsRequiredException, ResponseException, JsonProcessingException {
 
-	private void thenShouldCreateJiraVersion(Version version) throws CredentialsRequiredException, ResponseException, JsonProcessingException {
+        then(jiraService).should().addVersionToIssues(versionName, projectKey, ImmutableList.copyOf(issueKeys));
+    }
 
-		then(jiraService).should().createJiraVersion(version);
-	}
+    private void thenShouldCreateJiraVersion(String name, String project) throws CredentialsRequiredException, ResponseException, JsonProcessingException {
 
-	private void thenGetChangesets(VerificationMode verificationMode, String branchName) {
+        then(jiraService).should().createJiraVersion(new SerializedVersion(null, name, project, null, false));
```

Comment:
helper should be for creating a `SerializedVersion` from just `name` and `project` ... that way the interaction is more visible in the test method